java-checks-test-sources/src/main/java/checks/DateTimeFormatterMismatch.java

```diff
@@ -0,0 +1,103 @@
+    DateTimeFormatter.ofPattern("Y-ww", Locale.ENGLISH); // Compliant
+    DateTimeFormatter.ofPattern("YY-ww", Locale.ENGLISH); // Compliant
+    DateTimeFormatter.ofPattern("YYY-ww", Locale.ENGLISH); // Compliant
+    DateTimeFormatter.ofPattern("YYYY-ww", Locale.ENGLISH); // Compliant
+
+
+    //The bad stuff
+    DateTimeFormatter.ofPattern(NON_COMPLIANT_PATTERN); // Noncompliant
+    DateTimeFormatter.ofPattern("y-ww"); // Noncompliant
+    DateTimeFormatter.ofPattern("yy-ww"); // Noncompliant
+    DateTimeFormatter.ofPattern("yyy-ww"); // Noncompliant
+    DateTimeFormatter.ofPattern("yyyy-ww"); // Noncompliant
+    DateTimeFormatter.ofPattern(NON_COMPLIANT_PATTERN, Locale.ENGLISH); // Noncompliant
+    DateTimeFormatter.ofPattern("y-ww", Locale.ENGLISH); // Noncompliant
+    DateTimeFormatter.ofPattern("yy-ww", Locale.ENGLISH); // Noncompliant
+    DateTimeFormatter.ofPattern("yyy-ww", Locale.ENGLISH); // Noncompliant
+    DateTimeFormatter.ofPattern("yyyy-ww", Locale.ENGLISH); // Noncompliant
+
+
+    DateTimeFormatter.ofPattern("w"); // Compliant
+    DateTimeFormatter.ofPattern("ww"); // Compliant
+    DateTimeFormatter.ofPattern("w", Locale.ENGLISH); // Compliant
+    DateTimeFormatter.ofPattern("ww", Locale.ENGLISH); // Compliant
+
+
+    DateTimeFormatter.ofPattern("u-ww"); // Noncompliant
+    DateTimeFormatter.ofPattern("uu-ww"); // Noncompliant
+    DateTimeFormatter.ofPattern("uuuu-ww"); // Noncompliant
+    DateTimeFormatter.ofPattern("u-ww", Locale.ENGLISH); // Noncompliant
+    DateTimeFormatter.ofPattern("uu-ww", Locale.ENGLISH); // Noncompliant
+    DateTimeFormatter.ofPattern("uuuu-ww", Locale.ENGLISH); // Noncompliant
+  }
+
+  public void createUsingBuilder() {
+    new DateTimeFormatterBuilder()
+      .appendValue(ChronoField.YEAR, 4) // Noncompliant [[sc=20;ec=36;secondary=+2]] {{Change this year format to use the week-based year instead.}}
+      .appendLiteral('-')
+      .appendValue(WeekFields.ISO.weekOfWeekBasedYear(), 2)
+      .toFormatter();
+
+    new DateTimeFormatterBuilder()
+      .appendValue(WeekFields.ISO.weekBasedYear(), 4) // Noncompliant [[sc=20;ec=50;secondary=+2]] {{Change this year format to use ChronoField.YEAR instead.}}
+      .appendLiteral('-')
+      .appendValue(ChronoField.ALIGNED_WEEK_OF_YEAR, 2)
+      .toFormatter();
+
+    new DateTimeFormatterBuilder() // Compliant
+      .appendValue(WeekFields.ISO.weekBasedYear(), 4)
+      .appendLiteral('-')
+      .appendValue(WeekFields.ISO.weekOfWeekBasedYear(), 2)
+      .toFormatter();
+
+    new DateTimeFormatterBuilder() // Compliant
+      .appendValue(ChronoField.YEAR, 4)
+      .appendLiteral('-')
+      .appendValue(ChronoField.ALIGNED_WEEK_OF_YEAR, 2)
+      .toFormatter();
+
+    DateTimeFormatterBuilder builder = new DateTimeFormatterBuilder()
+      .appendValue(ChronoField.YEAR, 4) // Noncompliant [[sc=20;ec=36]]
+      .appendLiteral('-')
+      .appendValue(WeekFields.ISO.weekOfWeekBasedYear(), 2);
+
+    builder.toFormatter();
```

Comment:
For completeness, we could add an additional test case to document that we do not consider consecutive calls, which is ok:
```
DateTimeFormatterBuilder builder2 = new DateTimeFormatterBuilder();
builder2.appendValue(ChronoField.YEAR, 4); // FN - consecutive calls on builders are not taken into account
builder2.appendLiteral('-');
builder2.appendValue(WeekFields.ISO.weekOfWeekBasedYear(), 2);
builder2.toFormatter();
```